### PR TITLE
Fix a couple of bugs in the logic for how AWS metric periods are calculated

### PR DESCRIPTION
**Pull request by tommyers-elastic**
Firstly, we clarify that periods are always whole-minute durations.
Any period that is less than or in between minutes is rounded up to the
next whole-minute.

Secondly, we ensure that the resulting time period is always in the past.
This stops us getting empty metrics for the current-minute period.

Thirdly, we follow the AWS guidelines of aligning periods to whole
multiples within the hour e.g. 10:25->10:30 instead of 10:27->10:32
for a 5 minute period.

Fixes https://github.com/elastic/beats/issues/32687
Fixes https://github.com/elastic/beats/issues/32640

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: x-pack/metricbeat/module/aws/utils_test.go [hunk 1]
```diff
@@ -434,3 +434,54 @@ func TestGetResourcesTags(t *testing.T) {
...
+		start         string
+		period        time.Duration
+		latency       time.Duration
+		expectedStart string
+		expectedEnd   string
+	}{
+		// window should align with period e.g. requesting a 5 minute period at 10:27 gives 10:20->10:25
+		{"1 minute", "2022-08-15T13:38:45Z", time.Second * 60, 0, "2022-08-15T13:37:00Z", "2022-08-15T13:38:00Z"},
+		{"2 minutes", "2022-08-15T13:38:45Z", time.Second * 60 * 2, 0, "2022-08-15T13:36:00Z", "2022-08-15T13:38:00Z"},
+		{"3 minutes", "2022-08-15T13:38:45Z", time.Second * 60 * 3, 0, "2022-08-15T13:33:00Z", "2022-08-15T13:36:00Z"},
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/aws/utils_test.go:450]: just curious, why the start time and end time is not "2022-08-15T13:35:00Z", "2022-08-15T13:38:00Z"?

**Review by kaiyan-sheng — APPROVED** [review]
Overall the change looks good to me. One question: adding `now` as an input parameter for `GetStartTimeEndTime`, is it to make the tests easier?

**Comment by tommyers-elastic** [comment]
> Overall the change looks good to me. One question: adding `now` as an input parameter for `GetStartTimeEndTime`, is it to make the tests easier?

yep!

**Review by kaiyan-sheng — APPROVED** [review]
Could you add a changelog for this one please? Thanks!!

**Pull request merged**